apps/amo/decorators.py

```diff
@@ -238,3 +239,19 @@ def wrapper(request, username, *args, **kw):
             username = request.user.username
         return f(request, username, *args, **kw)
     return wrapper
+
+
+def atomic(fn):
+    """Set the transaction isolation level to SERIALIZABLE and then delegate
+    to transaction.atomic to run the specified code atomically. The
+    SERIALIZABLE level will run SELECTs in LOCK IN SHARE MODE when used in
+    conjunction with transaction.atomic.
+    Docs: https://dev.mysql.com/doc/refman/5.5/en/set-transaction.html.
+    """
+    @functools.wraps(fn)
+    def inner(*args, **kwargs):
+        cursor = connection.cursor()
```

Comment:
I guess this will only work when pinned to master? Maybe it doesn't matter. I'm not sure. Hmm.